src/ucp/wireup/wireup_ep.c

```diff
@@ -355,6 +355,7 @@ UCS_CLASS_INIT_FUNC(ucp_wireup_ep_t, ucp_ep_h ucp_ep)
     self->pending_count      = 0;
     self->flags              = 0;
     self->progress_id        = UCS_CALLBACKQ_ID_NULL;
+    self->cms_used_idx       = 0;
```

Comment:
UCP_NULL_RESOURCE

Reply:
why UCP_NULL_RESOURCE?
the index needs to start from the first cm on the list in index 0 - https://github.com/openucx/ucx/pull/5551/files#diff-36e34c1df2a6c773b9dda08dd8bf17eeR780

Reply:
at this point it's not "used" yet, need to initialize reference to valid object

Reply:
it's initialized to its firstly used value. same as we initialize counters and indexes...

Reply:
do you mean UCT EP is already created on CM by this index?
this is not a counter, this is index to associate 2 objects - EP and CM object

Reply:
at this point nothing was created. but after the wireup ep is created, the first value to use is 0.
instead of creating the ep and then initializing cm_idx to 0.
if this is the initial value of this index, why not set it here?

Reply:
because the link (CM EP to CM obj) is invalid at this point, again actually this is not a counter

Reply:
this index is the index of the cm to try to connect on. it doesn't mean that the cm ep was already created.
creating a wireup ep and then initializing this index to 0 would duplicate the code every time a wireup ep is created.

Reply:
not every wireup_ep uses this cm_idx, pls keep it initialized correctly, if it's not used it should be initialized by special value. If it's a counter call it like a retry counter or smth.